File: build-support/bin/xz/linux/x86_64/5.2.4-2/xz.py
#!/usr/bin/env python

import os
import subprocess
import sys

argv = sys.argv

this_file_dir = os.path.dirname(__file__)

xz_executable = os.path.realpath(os.path.join(this_file_dir, 'xz-real'))

xz_lib_dir = os.path.realpath(os.path.join('..', 'lib'))

platform = os.uname()[0]

if platform == 'Darwin':
  env_var = 'DYLD_LIBRARY_PATH'
elif platform == 'Linux':
  env_var = 'LD_LIBRARY_PATH'
else:
  raise ValueError('Unrecognized platform: {}.'.format(platform))

prev_lib_path = os.environ.get(env_var, '')
lib_path_entries = [s for s in prev_lib_path.split(':') if s != '']
lib_path_ours_first = [xz_lib_dir] + lib_path_entries

new_env = os.environ.copy()

new_env[env_var] = ':'.join(lib_path_ours_first)

# Inherit the standard fds, close any extra we may have opened.
rc = subprocess.call(sys.argv, executable=xz_executable, env=new_env, close_fds=True)
sys.exit(rc)

File: build-support/bin/xz/linux/x86_64/5.2.4-1/xz.pl
#!/usr/bin/perl

use strict;
use warnings;

use Cwd 'abs_path';
use File::Basename 'dirname';
use File::Spec::Functions 'catfile';
use POSIX 'uname';

my $this_file_dir = dirname(__FILE__);
my $xz_executable = abs_path(catfile($this_file_dir, 'xz-real'));
my $xz_lib_dir = abs_path(catfile($this_file_dir, '..', 'lib'));

my ($platform) = uname();

my $env_var;
if ($platform eq 'Darwin') {
  $env_var = 'DYLD_LIBRARY_PATH';
} elsif ($platform eq 'Linux') {
  $env_var = 'LD_LIBRARY_PATH';
} else {
  die sprintf("Unrecognized platform: {}", $platform);
}

my $prev_lib_path = $ENV{$env_var};
$prev_lib_path = '' unless defined $prev_lib_path;

my @lib_path_entries = grep(!/^$/m, split(/:/, $prev_lib_path));

my @lib_path_ours_first = ($xz_lib_dir, @lib_path_entries);

$ENV->{$env_var} = join(':', @lib_path_ours_first);

exec { $xz_executable } ($0, @ARGV);
